### PR TITLE
MT3-669 #ready-for-test Display current & previous post visit actions

### PR DESCRIPTION
**Pull request by Gweaton**
Previously submitted notes will be displayed in `Details` under the notes field. A note is also displayed on the Review page when a post visit action has been created.

### CODE REVIEW

File: components/PostVisitActionInput.tsx [hunk 1]
```diff
@@ -1,4 +1,7 @@
-import { Checkboxes as LBHCheckboxes } from "lbh-frontend-react/components";
+import {
+  Checkboxes as LBHCheckboxes,
+  List,
+} from "lbh-frontend-react/components";
 import { Textarea } from "lbh-frontend-react/components/Textarea";
```
Comment (erbridge) [inline components/PostVisitActionInput.tsx:5]: Next time you come across this sort of thing, combine them into a single import statement.

File: components/ReviewNotes.tsx [hunk 2]
```diff
@@ -14,7 +14,15 @@ export const ReviewNotes: React.FunctionComponent<ReviewNotesProps> = (
   return (
     <>
       {notes.map((note, i) => (
-        <div key={i}>{note.value}</div>
+        <div key={i}>
+          {note.value}
+          {note.isPostVisitAction &&
+            (note.createdAt ? (
+              <div>Post visit action created on {note.createdAt}.</div>
```
Comment (erbridge) [inline components/ReviewNotes.tsx:21]:
```suggestion
              <div>Post visit action created on {formatDate(note.createdAt, "d MMMM yyyy")}.</div>
```

File: components/PostVisitActionInput.tsx [hunk 3]
```diff
@@ -84,6 +92,20 @@ export const PostVisitActionInput: React.FunctionComponent<PostVisitActionInputP
...
+      {previousPostVisitActions.length > 0 && (
+        <Details
+          summary={{
+            id: "post-visit-actions",
+            value: "Previous post-visit-actions",
+          }}
+        >
+          <List
+            items={previousPostVisitActions.map(
+              (note) => `${note.value}. Created: ${note.createdAt}`
```
Comment (erbridge) [inline components/PostVisitActionInput.tsx:104]:
```suggestion
              (note) => `${note.value} (created on ${note.createdAt})`
```

We don't know what punctuation they'll use, so this feels cleaner.

Comment (erbridge) [reply components/PostVisitActionInput.tsx:104]: Or even better:

```suggestion
              (note) => `${note.createdAt}: ${note.value}`
```

**Review by erbridge — COMMENTED** [review]
Couple of small tweaks. Otherwise, looks good.

File: components/ReviewNotes.tsx (hunk 2 above)
Comment (Gweaton) [reply components/ReviewNotes.tsx:21]: I've actually stored it as a formatted string - should I undo that?

Comment (erbridge) [reply components/ReviewNotes.tsx:21]: Oh yes. We should be storing it as an ISO timestamp.

Comment (Gweaton) [reply components/ReviewNotes.tsx]: Nope, I'm being silly, we were doing that anyway - I just had to convert it into a date again to use `formatDate`

File: components/PostVisitActionInput.tsx [hunk 4]
```diff
@@ -84,6 +92,20 @@ export const PostVisitActionInput: React.FunctionComponent<PostVisitActionInputP
...
+      {previousPostVisitActions.length > 0 && (
+        <Details
+          summary={{
+            id: "post-visit-actions",
+            value: "Previous post-visit-actions",
+          }}
+        >
+          <List
+            items={previousPostVisitActions.map(
+              (note) => `${note.createdAt}: ${note.value}`
```
Comment (erbridge) [inline components/PostVisitActionInput.tsx:104]:
```suggestion
              (note) => `${formatDate(new Date(note.createdAt), "d MMMM yyyy")}: ${note.value}`
```

File: components/ReviewNotes.tsx [hunk 5]
```diff
@@ -14,7 +15,18 @@ export const ReviewNotes: React.FunctionComponent<ReviewNotesProps> = (
...
     <>
       {notes.map((note, i) => (
-        <div key={i}>{note.value}</div>
+        <div key={i}>
+          {note.value}
+          {note.isPostVisitAction &&
+            (note.createdAt ? (
+              <div>
+                Post visit action created on{" "}
+                {formatDate(new Date(new Date().toISOString()), "d MMMM yyyy")}.
```
Comment (erbridge) [inline components/ReviewNotes.tsx:24]:
```suggestion
                {formatDate(new Date(note.createdAt), "d MMMM yyyy")}.
```

**Review by erbridge — APPROVED** [review]

**Pull request merged**